components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java

```diff
@@ -1181,4 +1191,287 @@ private APIError handleNotFoundError() {
+                    .getResidentIdP(tenantDomain);
+            if (residentIdp != null) {
+                FederatedAuthenticatorConfig federatedAuthConfig = IdentityApplicationManagementUtil
+                        .getFederatedAuthenticator(residentIdp.getFederatedAuthenticatorConfigs(),
+                                IdentityApplicationConstants.Authenticator.SAML2SSO.NAME);
+                if (federatedAuthConfig == null) {
+                    throw handleException(Response.Status.NOT_FOUND,
+                            Constants.ErrorMessage.ERROR_CODE_FEDERATED_AUTHENTICATOR_CONFIG_NOT_FOUND,
+                            IdentityApplicationConstants.Authenticator.SAML2SSO.NAME);
+                }
+
+                Property[] idpProperties = federatedAuthConfig.getProperties();
+                if (idpProperties == null || idpProperties.length == 0) {
+                    throw handleException(Response.Status.NOT_FOUND,
+                            Constants.ErrorMessage.ERROR_CODE_FEDERATED_AUTHENTICATOR_PROPERTIES_NOT_FOUND,
+                            IdentityApplicationConstants.Authenticator.SAML2SSO.NAME);
+                }
+
+                inboundAuthConfig.setDestinationURLs(IdentityApplicationManagementUtil
+                        .getPropertyValuesForNameStartsWith(idpProperties,
+                                IdentityApplicationConstants.Authenticator.SAML2SSO.DESTINATION_URL_PREFIX));
+
+                String metadataValidityPeriod = IdentityApplicationManagementUtil.getPropertyValue(idpProperties,
+                        IdentityApplicationConstants.Authenticator.SAML2SSO.SAML_METADATA_VALIDITY_PERIOD);
+                if (StringUtils.isNotEmpty(metadataValidityPeriod)) {
+                    inboundAuthConfig.setMetadataValidityPeriod(Integer.parseInt(metadataValidityPeriod));
+                }
+
+                inboundAuthConfig.setEnableMetadataSigning(Boolean.parseBoolean(
+                        IdentityApplicationManagementUtil.getPropertyValue(idpProperties,
+                                IdentityApplicationConstants.Authenticator.SAML2SSO.SAML_METADATA_SIGNING_ENABLED)));
+
+                // Construct and set the SAML metadata download endpoint.
+                String samlMetadataEndpoint;
+                if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                    samlMetadataEndpoint = ServiceURLBuilder.create().addPath(
+                            Constants.SAML2_METADATA_ENDPOINT_URI_PATH).build().getAbsolutePublicURL();
+                } else {
+                    samlMetadataEndpoint = ServiceURLBuilder.create().addPath("/t/" + tenantDomain +
+                            Constants.SAML2_METADATA_ENDPOINT_URI_PATH).build().getAbsolutePublicURL();
+                }
+                inboundAuthConfig.setMetadataEndpoint(samlMetadataEndpoint);
+            } else {
+                throw handleException(Response.Status.NOT_FOUND,
```

Comment:
Not being able to find the resident IDP should be a server error IMO

Reply:
Should be same for similar errors in other places.

Reply:
Fixed